pkg/probes/health/grpc_health.go

```diff
@@ -18,8 +18,11 @@ func NewGrpcHealth(s *grpc.Server, addr net.Addr, timeout time.Duration, opts ..
 	return NewApplicationHealthFunc(
 		func() error {
 			ctx, closeFn := context.WithTimeout(context.Background(), timeout)
-			defer closeFn()
 			conn, err := tools.DialContext(ctx, addr, opts...)
+
+			defer closeFn()
```

Comment:
Why do we need to move this?

Reply:
Not 'need', but 'can' group the deferred functions together for best reading.